packages/components/src/components/post-icon/post-icon.tsx

```diff
@@ -1,37 +1,137 @@
 import { Component, Host, h, Prop, State } from '@stencil/core';
 
+const CDN_URL = 'https://unpkg.com/@swisspost/design-system-icons/public/post-icons';
+
+/**
+ * @class PostIcon - representing a stencil component
+ */
 @Component({
   tag: 'post-icon',
   styleUrl: 'post-icon.scss',
   shadow: true,
 })
 export class PostIcon {
-  @Prop() name: number;
+  /**
+   * The name/id of the icon (e.g. 1000, 1001, ...).
+   */
+  @Prop() name: string;
+
+  /**
+   * The base path, where the icons are located (must be a public url).
+   */
   @Prop() base?: string;
+
+  /**
+   * When set to `true`, the icon will be flipped horizontally.
+   */
+  @Prop() flipH?: boolean;
+
+  /**
+   * When set to `true`, the icon will be flipped vertically.
+   */
+  @Prop() flipV?: boolean;
+
+  /**
+   * The `number` for the css `scale` transformation.
+   */
+  @Prop() scale?: number;
+
+  /**
+   * The `number` of degree for the css `rotate` transformation.
+   */
+  @Prop() rotate?: number;
+
+  /**
+   * The name of the animation (`cylon`, `cylon-vertical`, `spin`, `spin-reverse`, `fade`, `throb`).
+   */
+  @Prop() animation?: string;
+  @State() initialPath: string;
   @State() path: string;
+  @State() svgSource: string = '<svg viewBox="0 0 16 16"></svg>';
+  @State() svgOutput: string;
 
-  constructor() {
+  connectedCallback() {
     // Construct icon path from different possible sources
     let basePath: string;
     const metaBase = document.head.querySelector(
       'meta[name="design-system-settings"][data-post-icon-base]',
     );
-    if (metaBase) basePath = metaBase.getAttribute('data-post-icon-base');
-    if (this.base) basePath = this.base;
-    if (!basePath) basePath = `https://unpkg.com/@swisspost/design-system-icons/public/svg`;
 
-    this.path = new URL(
+    if (this.base) {
+      basePath = this.base;
+    } else if (metaBase) {
+      basePath = metaBase.getAttribute('data-post-icon-base');
+    } else {
+      basePath = CDN_URL;
+    }
+
+    this.path = this.getPath(basePath);
+    this.svgSource = window.localStorage.getItem(`post-icon-${this.name}`) ?? this.svgSource;
+  }
+
+  componentWillLoad() {
+    this.fetchSVG();
+  }
+
+  componentWillRender() {
+    const svgStyles = Object.entries({
+      scale: this.scale && !isNaN(Number(this.scale)) ? `${this.scale}` : null,
```

Comment:
It would be good to have [prop validation](https://stenciljs.com/docs/properties#prop-validation)